tests/test_api.rs

```diff
@@ -3538,6 +3542,66 @@ fn inspector_dispatch_protocol_message() {
   assert_eq!(channel.count_flush_protocol_notifications, 0);
 }
 
+#[test]
+fn inspector_exception_thrown() {
+  let _setup_guard = setup();
+  let isolate = &mut v8::Isolate::new(Default::default());
+
+  use v8::inspector::*;
+  let mut default_client = ClientCounter::new();
+  let mut inspector = V8Inspector::create(isolate, &mut default_client);
+
+  let scope = &mut v8::HandleScope::new(isolate);
+  let context = v8::Context::new(scope);
+  let mut context_scope = v8::ContextScope::new(scope, context);
+
+  let name = b"";
+  let name_view = StringView::from(&name[..]);
+  inspector.context_created(context, 1, name_view);
+  let mut channel = ChannelCounter::new();
+  let state = b"{}";
+  let state_view = StringView::from(&state[..]);
+  let mut session = inspector.connect(1, &mut channel, state_view);
+  let message = String::from(
+    r#"{"id":1,"method":"Runtime.enable"}"#,
+  );
+  let message = &message.into_bytes()[..];
+  let string_view = StringView::from(message);
+  session.dispatch_protocol_message(string_view);
+  assert_eq!(channel.count_send_response, 1);
+  assert_eq!(channel.count_send_notification, 1);
+  assert_eq!(channel.count_flush_protocol_notifications, 0);
+
+  let message = "test exception".to_string();
+  let message = &message.into_bytes()[..];
+  let message_string_view = StringView::from(message);
+  let detailed_message = "detailed message".to_string();
+  let detailed_message = &detailed_message.into_bytes()[..];
+  let detailed_message_string_view = StringView::from(detailed_message);
+  let url = "file://exception.js".to_string();
+  let url = &url.into_bytes()[..];
+  let url_string_view = StringView::from(url);
+  let exception_msg = v8::String::new(&mut context_scope, "This is a test error").unwrap();
+  let exception = v8::Exception::error(&mut context_scope, exception_msg);
+  let stack_trace = v8::UniquePtr::from(V8StackTrace);
```

Comment:
That won't work - `V8StackTrace` can't be instantiated in Rust or C++, it's a virtual base class. Only V8 can hand them out.

You can get an instance from V8 through `V8Inspector::createStackTrace()` or `V8Inspector::captureStackTrace()`. The latter is straightforward to implement:
<details>

```diff
diff --git a/src/binding.cc b/src/binding.cc
index 176c95a..1246828 100644
--- a/src/binding.cc
+++ b/src/binding.cc
@@ -2056,6 +2056,11 @@ void v8_inspector__V8Inspector__contextCreated(
       ptr_to_local(&context), contextGroupId, humanReadableName));
 }
 
+v8_inspector::V8StackTrace* v8_inspector__V8Inspector__captureStackTrace(
+    v8_inspector::V8Inspector* self, bool full_stack) {
+  return self->captureStackTrace(full_stack).release();
+}
+
 void v8_inspector__V8InspectorSession__DELETE(
     v8_inspector::V8InspectorSession* self) {
   delete self;
diff --git a/src/inspector.rs b/src/inspector.rs
index b6cccd3..7957d4a 100644
--- a/src/inspector.rs
+++ b/src/inspector.rs
@@ -106,6 +106,10 @@ extern "C" {
     contextGroupId: int,
     humanReadableName: StringView,
   );
+  fn v8_inspector__V8Inspector__captureStackTrace(
+    this: *mut V8Inspector,
+    full_stack: bool,
+  ) -> *mut V8StackTrace;
 }
 
 #[no_mangle]
@@ -925,6 +929,17 @@ impl V8Inspector {
       )
     }
   }
+
+  pub fn capture_stack_trace(
+    &mut self,
+    full_stack: bool,
+  ) -> UniqueRef<V8StackTrace> {
+    unsafe {
+      UniqueRef::from_raw(v8_inspector__V8Inspector__captureStackTrace(
+        self, full_stack,
+      ))
+    }
+  }
 }
 
 impl Drop for V8Inspector {
```

</details>

Reply:
Thanks! I got stuck on this one and picked up something else :) that should get me of the ground

Reply:
I'm not really sure what's going on :/ failing on:
`thread 'inspector_exception_thrown' panicked at 'assertion failed: needs_drop::<T>()', src/support.rs:183:3`
I think this is coming from `UniqueRef` related to `V8StackTrace`

Reply:
```
fn assert_unique_ptr_layout_compatible<U, T>() {
  // Assert that `U` (a `UniqueRef` or `UniquePtr`) has the same memory layout
  // as a raw C pointer.
  assert_eq!(size_of::<U>(), size_of::<*mut T>());
  assert_eq!(align_of::<U>(), align_of::<*mut T>());

  // Assert that `T` (probably) implements `Drop`. If it doesn't, a regular
  // reference should be used instead of UniquePtr/UniqueRef.
  assert!(needs_drop::<T>());
}
```

So I guess we need to use a regular reference instead of `UniqueRef` :P

Reply:
> So I guess we need to use a regular reference instead of `UniqueRef` :P

I'm afraid it is the other way around: `Drop` needs to be implemented for `V8StackStrace`.